File: python/python_bandwidth_bxml/response.py
"""
response.py

Class that allows user to generate BXML programatically in python

@author Jacob Mulford
@copyright Bandwidth INC
"""

class Response():

    response_tag = "Response"
    xml_header = '<?xml version="1.0" encoding="UTF-8"?>'

    def __init__(self, verbs=[]):
        """
        Creates the Response class with an optional list of verbs

        :param list<Verb> verbs: optional ordered list of verbs to build the response
        """
        self.verbs = verbs

    def add_verb(self, verb):
        """
        Adds the Verb to the already existing verbs

        :param Verb verb: The Verb to add
        """
        self.verbs.push(verb)

    def to_xml(self):
        """
        Converts the Response class to its XML representation

        :rtype str: The XML representation of the Response class
        """
        xml_string = Response.xml_header
        xml_string += '<' + Response.response_tag + '>'
        for verb in self.verbs:
            xml_string += verb.to_xml()
        xml_string += '</' + Response.response_tag + '>' 

        return xml_string

File: nodejs/lib/response.js
/*
 * response.js
 *
 * This class is responsible for building BXML response objets. It is created
 * by attaching various BXML verbs
 *
 * @author Jacob Mulford
 * @copyright Bandwidth INC
 */

var xmlbuilder = require("xmlbuilder");

class Response {
    constructor() {    
      this.verbs = [];
    }

    /**
     * Converts the Response object into an XML string based on its verbs
     *
     * @return {String} The XML representation of the object
     */
    toXml() {
        var xml = xmlbuilder.create("Response", {
            version: "1.0",
            encoding: "UTF-8"
        })

        for (var i = 0; i < this.verbs.length; i++) {
            xml = this.verbs[i].addXml(xml);
        }
        return xml.end();
    }

    /**
     * Adds the Verb object to the list of verbs
     *
     * @param {Verb} verb The verb object to add to the list of verbs
     */
    addVerb(verb) {
        this.verbs.push(verb);
    }
}

module.exports = Response;
